util/etcd/etcd.go

```diff
@@ -0,0 +1,382 @@
+	}
+	defer cli.Close()
+	var lastError error
+	var resp *clientv3.MemberListResponse
+	err = wait.ExponentialBackoff(c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		resp, err = cli.MemberList(ctx)
+		cancel()
+		if err == nil {
+			return true, nil
+		}
+		c.log.Error(err, "failed to get etcd member list")
+		lastError = err
+		return false, nil
+	})
+	if err != nil {
+		return nil, lastError
+	}
+
+	alarms, err := c.Alarms()
+	if err != nil {
+		return nil, err
+	}
+
+	members := make([]Member, len(resp.Members))
+	for i, m := range resp.Members {
+		newMember := clientMemberToclientMember(m)
+		for _, c := range *alarms {
+			if c.MemberID == newMember.ID {
+				newMember.Alarms = append(newMember.Alarms, c.Type)
+			}
+			members[i] = newMember
+		}
+	}
+
+	return &members, nil
+}
+
+func clientMemberToclientMember(m *etcdserverpb.Member) Member {
+	return Member{
+		ID:         m.GetID(),
+		Name:       m.GetName(),
+		PeerURLs:   m.GetPeerURLs(),
+		ClientURLs: m.GetClientURLs(),
+		IsLearner:  m.GetIsLearner(),
+		Alarms:     []AlarmType{},
+	}
+}
+
+// Member retrieves the Member information for a given peer
+func (c *Client) Member(peerURL string) (*Member, error) {
+	members, err := c.Members()
+	if err != nil {
+		return nil, err
+	}
+	for _, m := range *members {
+		if m.PeerURLs[0] == peerURL {
+			return &m, nil
+		}
+	}
+	return nil, errors.New("NotFound")
+}
+
+// Status returns the cluster status as known by the connected etcd server
+func (c *Client) Status() (*Status, error) {
+	cli, err := c.newEtcdClient()
+	if err != nil {
+		return nil, err
+	}
+	var lastError error
+	var resp *clientv3.StatusResponse
+	err = wait.ExponentialBackoff(c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		resp, err = cli.Status(ctx, c.endpoint)
+		cancel()
+		if err == nil {
+			return true, nil
+		}
+		c.log.Error(err, "failed to get etcd status")
+		lastError = err
+		return false, nil
+	})
+	if err != nil {
+		return nil, lastError
+	}
+
+	status := Status{
+		Leader:           resp.Leader,
+		Version:          resp.Version,
+		Errors:           resp.Errors,
+		IsLearner:        resp.IsLearner,
+		RaftIndex:        resp.RaftIndex,
+		RaftTerm:         resp.RaftTerm,
+		RaftAppliedIndex: resp.RaftAppliedIndex,
+		DbSizeInUse:      resp.DbSizeInUse,
+		DbSize:           resp.DbSize,
+	}
+
+	return &status, nil
+}
+
+// MoveLeader moves the leader to the provided member ID
+func (c *Client) MoveLeader(transfereeID uint64) error {
+	cli, err := c.newEtcdClient()
+	if err != nil {
+		return err
+	}
+	var lastError error
+	err = wait.ExponentialBackoff(c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		_, err = cli.MoveLeader(ctx, transfereeID)
+		cancel()
+		if err == nil {
+			return true, nil
+		}
+		c.log.Error(err, "failed to get move etcd member")
```

Comment:
I might be wrong on this suggestion depending on what `MoveLeader` does

```suggestion
		c.log.Error(err, "failed to move etcd member")
```

Reply:
I think @randomvariable meant "failed to move etcd leader," because it's not possible to move an etcd member :)